frontend/app/model/Nav.scala

```diff
@@ -10,13 +10,8 @@ object Nav {
     subNav: Seq[NavItem] = Nil
   )
 
-  def fetchNav(url: String) = {
-    val navigationURLs = navigation.find { navItem =>
-      // Return the parent item if URL is a subnav
-      navItem.href == url || navItem.subNav.map(_.href).contains(url)
-    }
-    navigationURLs
-  }
+  def fetchNav(url: String) =
+    navigation.find(navItem => url.contains(navItem.href))
```

Comment:
Previous logic prevented us showing the secondary nav in the list view as it doesn't need to be in the nav.

This version just matches against the url. In this case anything under `/whats-on` will return the what's on parent nav.